src/github.com/getlantern/fronted/dialer.go

```diff
@@ -212,6 +213,9 @@ func (ddf *DirectDomainTransport) RoundTrip(req *http.Request) (resp *http.Respo
 	if err != nil {
 		return nil, fmt.Errorf("Unable to construct request for url '%s' with error '%v'", normalized, err)
 	}
+	if err = deepcopy.Copy(&norm.Header, req.Header); err != nil {
+		return nil, fmt.Errorf("Unable to copy request headers: %v", err)
+	}
 	return ddf.Transport.RoundTrip(norm)
```

Comment:
I just realized there's a better way to do this whole request copying business -- I'll submit another PR.